src/main/java/com/kickstarter/dropwizard/metrics/influxdb/InfluxDbMeasurement.java

```diff
@@ -147,34 +152,85 @@ public Builder putTag(final String key, final String value) {
     /**
      * Adds the given key-value pair to the fields map.
      *
-     * @throws IllegalArgumentException if any value is not a String or primitive.
+     * @throws IllegalArgumentException if any value is not a String, primitive,
+     *         or Collection of Strings and primitives.
      */
     public <T> Builder putField(final String key, final T value) {
-      if (value == null) {
-        return this;
+      if (value instanceof Collection<?>) {
+        final Collection collection = (Collection) value;
+        final String collString = validatedPrimitiveCollection(key, collection);
+        fields.put(key, collString);
+      } else if (value != null) {
+        final Optional<String> fieldString = validatedPrimitiveField(key, value);
+        fieldString.ifPresent(s -> fields.put(key, s));
+      }
+
+      return this;
+    }
+
+    /**
+     * Validates that the collection only contains Strings and primitives.
+     *
+     * @return the collection as a string, if valid.
+     * @throws IllegalArgumentException if any collection value is not a string or primitive.
+     */
+    private static String validatedPrimitiveCollection(final String key, final Collection collection) {
+      for (final Object value : collection) {
+        if (!isValidField(value)) {
+          throw new IllegalArgumentException(
+            String.format(
+              "InfluxDbMeasurement collection field '%s' must contain only Strings and primitives: invalid field '%s'",
+              key, value
+            )
+          );
+        }
       }
 
+      return collection.toString();
+    }
+
+    /**
+     * Validates that the field is a String or primitive.
+     *
+     * @return true if the field is a String or primitive.
+     */
+    private static <T> boolean isValidField(final T value) {
+      return VALID_FIELD_CLASSES.stream().anyMatch(klass -> klass.isInstance(value));
```

Comment:
`Float`, `Double`, `Integer`, `Long`, `String`, and `Boolean` are all final classes, so it shouldn't make difference between the `isInstance` and identity check. Since we use generics in the `isValidField` method and Java don't support primitives in collections, we also don't have use cases when we compare a boxed class with a primitive class (`Double.class` and `double.class`). So, I think we could declare `VALID_FIELD_CLASSES` as an immutable set and use the `contains` method which should provide a faster lookup.

Reply:
Good call; made these changes in 5d4b72c.